src/Extensions/Abstractions/Activities/ActivityHacks.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace Microsoft.Omex.Extensions.Abstractions.Activities
+{
+	internal static class ActivityHacks
+	{
+		/// <summary>
+		/// Adds or replaces baggage item in the Activity
+		/// </summary>
+		/// <remarks>
+		/// The method created as a temporary replacement until actual SetBaggage will be available, most probably in net6.0 (https://github.com/dotnet/runtime/issues/42706)
+		/// It relies heavily on the internal structure of the Activity and might need updates after activity implementation change (if it's still needed in net6.0)
+		/// It's named the same as future API method to force compiler error (or automatic usage of proper method) when the actual method available and force deletion of this hack
+		/// </remarks>
+		internal static Activity SetBaggage(this Activity activity, string key, string value)
+		{
+			bool needToAddKey = true;
+
+			if (activity.GetBaggageItem(key) != null)
+			{
+				needToAddKey = !TryReplaceBaggageValue(activity, key, value);
+			}
+
+			if (needToAddKey)
+			{
+				activity.AddBaggage(key, value);
+			}
+
+			return activity;
+		}
+
+		private static bool TryReplaceBaggageValue(Activity activity, string key, string value)
+		{
+			s_baggageField ??= activity.GetType().GetField("_baggage", BindingFlags.NonPublic | BindingFlags.Instance);
+			if (s_baggageField == null)
+			{
+				return false;
+			}
+
+			object? baggage = s_baggageField.GetValue(activity);
+
+			s_headField ??= s_baggageField.FieldType.GetField("_first", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
`s_headField` field name vs `"_first"` field name

Reply:
To me, `head` makes more sense since it's the begging of the LinkedList, I don't sure that renaming would add more clarity (ex. `s_firstField` looks confusing)

Reply:
Mainly thinking since it's 100% coupled to the internals.